### PR TITLE
Use datetime-local in datetime form fields

### PR DESCRIPTION
**Pull request by skulonen**
# Description

**What?**

This PR adds a datetime picker to all datetime fields in forms.

**Why?**

This was mentioned when discussing the deviation UI with Aalto teachers.

**How?**

A new widget, `DateTimeLocalInput`, has been added. This new widget is otherwise the same as the built-in `DateTimeInput`, except it uses the `datetime-local` HTML input type, which uses a datetime picker provided by the browser. `datetime-local` expects the initial value of the field to be in a specific format, which the widget ensures. `datetime-local` is supported in all major browsers except for IE. If the user's browser doesn't support this widget type, it uses `text` instead (which the A+ forms used before this update anyway).

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [x] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [x] Django unit tests.
- [x] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Tested manually that the initial value is populated to the field correctly (in the course configuration form). Tested that the entered value is saved correctly.

**Did you test the changes in**

- [x] Chrome
- [x] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [x] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -1382,7 +1382,7 @@ msgstr "New deadline"
 
 #: deviations/forms.py
 msgid "DEVIATION_NEW_DEADLINE_DATE_HELPTEXT"
-msgstr "New submission deadline in the future in format YYYY-MM-DD HH:MM."
+msgstr "New submission deadline in the future."
```
Comment (Mankro) [inline locale/en/LC_MESSAGES/django.po:1385]:
```suggestion
msgstr "New submission deadline."
```
I don't think it is necessary to emphasize "future" here. If the user tries to enter an invalid date in the past, then the form validation should show a descriptive error.

File: locale/fi/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -1386,7 +1386,7 @@ msgstr "Uusi määräaika"
 
 #: deviations/forms.py
 msgid "DEVIATION_NEW_DEADLINE_DATE_HELPTEXT"
-msgstr "Uusi määräaika formaatissa YYYY-MM-DD HH:MM."
+msgstr "Uusi määräaika tulevaisuudessa."
```
Comment (Mankro) [inline locale/fi/LC_MESSAGES/django.po:1389]:
```suggestion
msgstr "Uusi määräaika."
```

**Review by Mankro — CHANGES_REQUESTED** [review]
Awesome! I just thought that the label text could be more compact.

**Comment by skulonen** [comment]
I've shortened the labels as you wished.

> If the user tries to enter an invalid date in the past, then the form validation should show a descriptive error.

I agree, but I don't think this is directly related to this PR. There are other issues related to that form's validation, too: it doesn't check that you've entered at least one exercise or module, and when you haven't entered a date or a minute number, it sends you back to the deviation list and shows the error message at the top of the page. These issues could be addressed in PR #401 instead.

**Comment by skulonen** [comment]
I rebased this and force-pushed.

**Review by Mankro — APPROVED** [review]

**Pull request merged**